src/main/java/gmail/filters/EmailProcessor.java

```diff
@@ -0,0 +1,22 @@
+package gmail.filters;
+
+import lombok.NonNull;
+
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+public class EmailProcessor {
+    public void processEmails(@NonNull List<Email> emails,
+                              @NonNull Predicate<Email> predicateFilter,
+                              @NonNull Consumer<Email> consumerFilter,
+                              @NonNull Function<Email, String> functionFilter) {
+        emails.forEach(email -> {
+            if (predicateFilter.test(email)) {
+                consumerFilter.accept(email);
+                functionFilter.apply(email);
```

Comment:
смотри, accept выводит на экран сообщение о том, что мы обработали сообщение
но по факту обрабатываешь его ты только на следующей строчке хехе

еще: у тебя функция в мейне берет тело имейла, который ты передаешь и переводит его в upper case  
если ты передаешь исходный имейл, мы теряем его изначальное состояние после того, как сработала функция 

Это значит, что пользовать никогда больше не сможет вернуться к исходному письму 
На почте так, конечно, делать нельзя) 
давай подумаем, как не потерять изначальное сообщение

Reply:
Не совсем понял почему теряется исходное состояние имейла, ведь функция не меняет само тело исходного, а только возвращает преобразованное тело, сам имейл не изменяется. Сигнатура функции же Function<Email, String>.
Я могу переписать с Function<Email, Email>
```suggestion
                consumerFilter.accept(functionFilter.apply(email));
```
Например с такой функцией
```suggestion
                email -> new Email(email.getSubject(), email.getBody().toUpperCase(), email.isImportant())
```
Типо такого? Или я не понял?